### ISSUE TITLE
Tfsec want's us to write access logs for access logs bucket, do we need it?

### ISSUE THREAD

**Issue by Andrey9kin** [fivexl/terraform-aws-account-baseline#4]
The problem we are trying to solve is the fact that we need to protect access logs or at least know if they were deleted

We are going to create access logs buckets (one per supported region since not every service can write cross-account) in the logs account and setup bucket policy that will only allow writing logs from that particular environment thus protecting access logs from deletion by attacker or accidental deletion

We still need to protect access logs in logs account somehow. Object lock or AWS Backup vault lock should be the go to option

Thus resolution is not to create access logs for access logs bucket but use protections outlined above.

We need to update access bucket module to make sure we can create it conditionally, so it have a bucket policy and eventually add object lock or aws backup (create a separate issue for it and link here)

**Comment by EreminAnton** [comment]
We cant use ObjectLock on access logs destination bucket:
[S3 buckets that have S3 Object Lock enabled can't be used as destination buckets for server access logs. Your destination bucket must not have a default retention period configuration.](https://docs.aws.amazon.com/AmazonS3/latest/userguide/enable-server-access-logging.html#:~:text=S3%20buckets%20that%20have%20S3%20Object%20Lock%20enabled%20can%27t%20be%20used%20as%20destination%20buckets%20for%20server%20access%20logs.%20Your%20destination%20bucket%20must%20not%20have%20a%20default%20retention%20period%20configuration.)

And looks like AWS Backup vault lock is not solving problem of deletion logs from the bucket, it solving problem of backup deletion, and only after files were backuped(Maybe if there an option to instantly backup every file right after placement it would be a solution)

Will look more in to it

**Comment by Andrey9kin** [comment]
Logs will be synced somehow to logs archive so protection of local bucket is less of an issue
